app/src/main/java/de/westnordost/streetcomplete/data/osm/mapdata/MapDataController.kt

```diff
@@ -260,3 +279,14 @@ class MapDataController internal constructor(
         private const val TAG = "MapDataController"
     }
 }
+
+// StyleableOverlayManager loads z16 tiles, but we want smaller tiles. Small tiles make db fetches for
+// typical getMapDataWithGeometry calls noticeably faster than z16, as they usually only require a small area.
+private const val SPATIAL_CACHE_TILE_ZOOM = 17
+
+// Twice the maximum number of tiles that can be loaded at once in StyleableOverlayManager (translated from z16 tiles).
+// We don't want to drop tiles from cache already when scrolling the map just a bit!
+private const val SPATIAL_CACHE_TILES = 128
+
+// In a city this is the approximate number of nodes in ~4-8 z16 tiles
+private const val SPATIAL_CACHE_INITIAL_CAPACITY = 20000
```

Comment:
FYI this initial capacity should not be "what can be usually expected" but rather the upper limit or more of what can be expected. Because if it is the former, it means that in 50% of the cases, the capacity needs to be doubled (automatically by the hashmap)

Reply:
So we have 32 z16 tiles, we would need some 80-160k.

Actually, is there any drawback if the size is chosen too large? After all, not everyone will use SC in a city.

Reply:
The hashmap will be bigger. Default load factor is 0.75, so an (empty) hashmap with 20000 entries should need about `20000 / 0.75 * 8 = 213 kilobytes` (1 reference should be 8 bytes)

Reply:
Ok, so switching to 100000 may be fine...
If the user is not inside a huge and very well mapped city, that should not be exceeded.

Reply:
Switched to 100k